src/port1.0/portconfigure.tcl

```diff
@@ -532,7 +532,12 @@ proc portconfigure::configure_get_universal_ldflags {} {
 
 # internal proc to determine if the compiler supports -arch
 proc portconfigure::arch_flag_supported {compiler} {
-    return [regexp {^gcc-4|llvm|apple|clang} $compiler]
+    # GCC prior to 4.7 does not accept -arch flag
+    if {[regexp {^macports(?:-[^-]+)?-gcc-4\.[0-6]} $compiler]} {
+        return 0
```

Comment:
1, I  mean yes, booleans should be `yes`/`no`.
Hopefully, this is fixed in 0d59ef0dc98d3ed4b4fee45e82896920123b1dfa.

Reply:
While newer gcc do accept -arch, they only accept *one*.  We need to be careful that this function isn't used elsewhere with the assumption that multiple -arch flags work.

Reply:
Hopefully, bc5b48d4972adff2943d0256dc583a3b3ca994d9 makes clear the difference between accepting one and multiple `-arch` flags.
Just accepting a single `-arch` is the default since that is how it is used in portconfigure.tcl.